client/lib/cart-values/cart-items.ts

```diff
@@ -146,6 +146,17 @@ export function hasDomainRenewal( cart: ResponseCart ): boolean {
 	return getAllCartItems( cart ).some( isDomainRenewal );
 }
 
+export function hasMarketplaceProduct( cart: ResponseCart ): boolean {
+	return getAllCartItems( cart ).some( isMarketplaceProduct );
+}
+
+/**
+ * @todo Confirm the condition below
+ */
+function isMarketplaceProduct( cartItem: ResponseCartProduct ): boolean {
+	return cartItem.cost > 0;
```

Comment:
All (or most) items in cart would have `cost > 0` but not all of them are Marketplace products (eg Plans, Domains). Have you tried using `isMarketplaceProduct`?

https://github.com/Automattic/wp-calypso/blob/b20b9e7b48aed5a40352f9eb4fde81dc7c9ad031/client/state/products-list/selectors/is-marketplace-product.js#L5-L8